compiler/rustc_mir_transform/src/gvn.rs

```diff
@@ -1407,6 +1419,28 @@ impl<'body, 'tcx> VnState<'body, 'tcx> {
             }
         }
 
+        // Aggregate-then-Transmute can just transmute the original field value,
+        // so long as the type is transparent over only that one single field.
+        if let Transmute = kind
+            && let Value::Aggregate(
+                AggregateTy::Def(aggregate_did, aggregate_args),
+                FIRST_VARIANT,
+                field_values,
+            ) = self.get(value)
+            && let [single_field_value] = **field_values
+            && let adt = self.tcx.adt_def(aggregate_did)
+            && adt.is_struct()
+            && adt.repr().transparent()
+        {
+            let field_ty = adt.non_enum_variant().single_field().ty(self.tcx, aggregate_args);
```

Comment:
Something's bothering me about this new opt, i'll need to check out the PR instead of just looking at it on my phone with my 4 months out of date memory of how things work. I'll look at it next week

Reply:
Gave it a look and can't find anything wrong with it.